fiftyone/server/routes/embeddings_v2.py

```
@@ -0,0 +1,778 @@
+def _slice(array, data):
+    """Applies a request's optional wire-order ``offset``/``limit``."""
+    offset = data.get("offset", None)
+    limit = data.get("limit", None)
+    if offset is None and limit is None:
+        return array
+
+    start = int(offset or 0)
+    length = int(limit) if limit is not None else None
+    # Negative values would silently slice from the wrong end
+    if start < 0 or (length is not None and length < 0):
+        raise ValueError("offset and limit must be non-negative")
```

Ruff (0.15.21)

[warning] Avoid specifying long messages outside the exception class

(TRY003)